dev/bots/prepare_package.dart

```diff
@@ -0,0 +1,145 @@
+  int errorCode = 0;
+
+  @override
+  String toString() => message;
+}
+
+class ArchivePreparer {
+  ArchivePreparer(String flutterRoot, String tempDir, {ProcessRunner runner})
+      : flutterRoot = path.absolute(flutterRoot),
+        tempDir = path.absolute(tempDir),
+        flutter = path.join(flutterRoot, 'bin', Platform.isWindows ? 'flutter.bat' : 'flutter'),
+        _runner = runner ?? Process.runSync {
+    environment = new Map<String, String>.from(Platform.environment);
+    environment['PUB_CACHE'] = path.join(flutterRoot, '.pub-cache');
+  }
+
+  final String flutterRoot;
+  final String tempDir;
+  final String flutter;
+  final ProcessRunner _runner;
+  final String git = Platform.isWindows ? 'git.bat' : 'git';
+  Map<String, String> environment;
+
+  String runProcess(String executable, List<String> args) {
+    stderr.write('Running "$executable ${args.join(' ')}" in $flutterRoot.\n');
+    final ProcessResult result = _runner(
+      executable,
+      args,
+      workingDirectory: flutterRoot,
+      environment: environment,
+    );
+    stdout.write(result.stdout);
+    stderr.write(result.stderr);
+    if (result.exitCode != 0) {
+      final String message =
+          'Running "$executable ${args.join(' ')}" in $flutterRoot failed with ${result.exitCode}.';
+      stderr.write('$message\n');
+      throw new ProcessFailedException(message);
+    }
+    return result.stdout.trim();
+  }
+
+  String runFlutter(List<String> args) {
+    return runProcess(flutter, args);
+  }
+
+  String runGit(List<String> args) {
+    return runProcess(git, args);
+  }
+
+  void prepareForArchive() {
+    // We want to user to start out the in the 'master' branch instead of a
+    // detached head. To do that, we need to make sure master points at the
+    // right (HEAD) revision.
+    final String revision = runGit(<String>['rev-parse', 'HEAD']);
+    runGit(<String>['checkout', 'master']);
+    runGit(<String>['reset', '--hard', revision]);
+
+    // Make the origin point to github instead of the chromium mirror.
+    runGit(<String>['remote', 'remove', 'origin']);
+    runGit(<String>['remote', 'add', 'origin', 'https://github.com/flutter/flutter.git']);
+
+    runFlutter(<String>['update-packages']);
+    runFlutter(<String>['doctor']);
+    runFlutter(<String>['precache']);
+    runFlutter(<String>['ide-config']);
+
+    // Create each of the templates, since they will call pub get on
+    // themselves when created, and this will warm the cache with their
+    // dependencies too.
+    for (String template in <String>['app', 'package', 'plugin']) {
+      final String createName = path.join(tempDir, 'create_$template');
+      runFlutter(
+        <String>['create', '--template=$template', createName],
+      );
+    }
+
+    // Yes, we could just skip all .packages files for the archive, but some
+    // are checked in, and we don't want to skip those.
+    runGit(<String>['clean', '-f', '-X', '**/.packages']);
```

Comment:
please warn at the top of this script that this script is going to delete your files :-)

Reply:
in fact this whole script should probably refuse to run locally, given how destructive it can be and how much it depends on the mint condition of your repo